tests/components/bsblan/test_water_heater.py

```diff
@@ -0,0 +1,253 @@
+    DOMAIN as WATER_HEATER_DOMAIN,
+    STATE_ECO,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.const import ATTR_ENTITY_ID, ATTR_TEMPERATURE, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.entity_registry as er
+
+from . import setup_with_selected_platforms
+
+from tests.common import (
+    MockConfigEntry,
+    async_fire_time_changed,
+    load_json_object_fixture,
+    snapshot_platform,
+)
+
+ENTITY_ID = "water_heater.bsb_lan"
+
+
+@pytest.fixture
+def mock_bsblan_with_methods(mock_bsblan):
+    """Create a mock with proper method signatures."""
+    # Create an async mock for set_hot_water with the correct signature
+    mock_bsblan.set_hot_water = AsyncMock()
+    return mock_bsblan
+
+
+@pytest.mark.parametrize(
+    ("dhw_file"),
+    [
+        ("dhw_state.json"),
+    ],
+)
+async def test_water_heater_states(
+    hass: HomeAssistant,
+    mock_bsblan_with_methods: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    dhw_file: str,
+) -> None:
+    """Test water heater states with different configurations."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+async def test_water_heater_entity_properties(
+    hass: HomeAssistant,
+    mock_bsblan_with_methods: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test the water heater entity properties."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    state = hass.states.get(ENTITY_ID)
+    assert state is not None
+
+    # Test when nominal setpoint is "10"
+    mock_setpoint = MagicMock()
+    mock_setpoint.value = 10
+    mock_bsblan_with_methods.hot_water_state.return_value.nominal_setpoint = (
+        mock_setpoint
+    )
+
+    freezer.tick(timedelta(minutes=1))
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+
+    state = hass.states.get(ENTITY_ID)
+    assert state.attributes.get("temperature") == 10
+
+
+@pytest.mark.parametrize(
+    ("mode", "bsblan_mode"),
+    [
+        (STATE_ECO, "Eco"),
+        (STATE_OFF, "Off"),
+        (STATE_ON, "On"),
+    ],
+)
+async def test_set_operation_mode(
+    hass: HomeAssistant,
+    mock_bsblan_with_methods: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    mode: str,
+    bsblan_mode: str,
+) -> None:
+    """Test setting operation mode."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    await hass.services.async_call(
+        WATER_HEATER_DOMAIN,
+        "set_operation_mode",
+        {
+            ATTR_ENTITY_ID: ENTITY_ID,
+            "operation_mode": mode,
```

Comment:
can we use constantsfor the service name and the operation mode?

Reply:
missed this one. Try to fix it now